### PR TITLE
feat(add-ons): add-on dependencies should be automatically downloaded, fixes #5337

### LINKED ISSUE #5337: Allow add-ons to download other add-ons as dependencies
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

In addition to https://github.com/ddev/ddev/issues/4717 I'd like the ability to download dependencies automatically.

If you have multiple dependencies (e.g. in a metapackage) it is tedious that you must manually install and update the dependencies.

### Describe your solution

Therefore, I'd like some global setting "allow automatic add-on dependency management" and corresponding features.

### Describe alternatives

Since DDEV is heavily focused on PHP it might be an option to use an established package manager for it like https://getcomposer.org/

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rfay**
## The Issue

- #5337

This PR 
- implements automatic addon dependency resolution and installation Previously, users had to manually install dependencies in the correct order, which was error-prone and led to installation failures.
- Allows an add-on to dynamically add additional dependencies which may be discovered during installation. For example `ddev-upsun` discovers needs for extra services during parsing of the `.upsun/config.yaml` so must inject them into the post-install.

## How This PR Solves The Issue

This PR introduces comprehensive automatic dependency resolution for DDEV addons through several key enhancements:

## TODO

- [x] Careful review of code. 
- [x] Make sure this solves the need of ddev-upsun for automatic dependency addition
- [x] Verify that docs reflect the changes completely
- [x] Check docs about dynamic dependency addition

## Docs

Rendered at https://ddev--7586.org.readthedocs.build/en/7586/

Key items:
* https://ddev--7586.org.readthedocs.build/en/7586/users/extend/creating-add-ons/
* https://ddev--7586.org.readthedocs.build/en/7586/users/usage/commands/#add-on

### Core Changes

1. **Automatic Dependency Installation**: When installing an addon with dependencies, DDEV now automatically installs missing dependencies recursively instead of failing with an error.

2. **Circular Dependency Detection**: Robust detection prevents infinite loops when addons have circular dependencies.

3. **Runtime Dependency Discovery**: New capability for addons to discover and declare dependencies dynamically during pre-install actions by creating `.runtime-deps-{addon-name}` files. This will be critical for the `ddev-upsun` add-on, which needs to install add-ons to parallel Upsun features.

4. **Relative Path Support**: Enhanced dependency resolution supporting relative and absolute paths (e.g., `../addon-name`) in addition to GitHub repos.

5. **GitHub API Integration**: Extracted reusable `GetGitHubRelease` function for consistent GitHub addon installation across dependency resolution.

6. **Skip Deps Flag**: Added `ddev add-on install <add-on> --skip-deps` flag to for users who want to skip automatic dependency installation and handle dependencies manually.

### Backward Compatibility

- All existing addon functionality preserved - no breaking changes
- Existing addons continue to work without modification
- New features are additive and optional
- Function signatures maintained for external usage

## Manual Testing Instructions

1. **Test Basic Dependency Installation**:
   ```bash
   # This should automatically install mock-redis as a dependency
   ddev add-on get /path/to/cmd/ddev/cmd/testdata/TestRealWorldDependencies/mock_redis_commander -v
   
   # Verify both addons are installed
   ddev add-on list --installed
   ls .ddev/docker-compose.redis*.yaml
   ```

2. **Test Circular Dependency Detection**:
   ```bash
   # This should fail with circular dependency error
   ddev add-on get /path/to/cmd/ddev/cmd/testdata/TestAddonCircular/addon_x -v
   ```

3. **Test --skip-deps Flag**:
   ```bash
   # Remove previously installed addons
   ddev add-on remove mock-redis-commander
   ddev add-on remove mock-redis
   
   # This should fail without dependency
   ddev add-on get --skip-deps /path/to/cmd/ddev/cmd/testdata/TestRealWorldDependencies/mock_redis_commander
   
   # Install dependency manually, then retry
   ddev add-on get /path/to/testdata/TestRealWorldDependencies/mock_redis
   ddev add-on get --skip-deps /path/to/cmd/ddev/cmd/testdata/TestRealWorldDependencies/mock_redis_commander
   ```

4. **Test Runtime Dependencies**:
   ```bash
   # This addon dynamically discovers dependencies during pre-install
   ddev add-on get /path/to/cmd/ddev/cmd/testdata/TestAddonWithRuntimeDeps/runtime_deps_addon -v
   
   # Should see "Installing runtime dependencies" in output
   # Should install mock-redis and mock-redis-commander automatically
   ddev add-on list --installed
   ```

### Test Existing Addon Compatibility

6. **Verify No Breaking Changes**:
   ```bash
   # Test that existing GitHub addons still work
   ddev add-on get ddev/ddev-redis -v
   ddev add-on remove ddev-redis
   
   # Test existing addon removal
   ddev add-on remove mock-redis-commander
   ddev add-on remove mock-redis
   ```
## Automated Testing Overview

This PR introduces comprehensive test coverage for the new dependency functionality:

### New Test Functions

1. **TestAddonGetWithDependencies**: Tests automatic installation of addons with static dependencies
2. **TestAddonGetCircularDependencies**: Verifies circular dependency detection and rejection
3. **TestAddonGetSkipDepsFlag**: Tests the `--skip-deps` flag behavior
4. **TestAddonGetRuntimeDependencies**: Tests runtime dependency discovery and installation

### New Test Structure

- **TestRealWorldDependencies/**: Mock Redis and Redis Commander addons with realistic dependency relationships
- **TestAddonCircular/**: Three addons (addon_x, addon_y, addon_z) with circular dependencies for testing detection
- **TestAddonWithRuntimeDeps/**: Runtime dependency addon that creates `.runtime-deps` file during pre-install

### Enhanced Test Coverage

- **ParseRuntimeDependencies**: Tests parsing of runtime dependency files with comments, empty lines, and various formats
- **GetGitHubRelease**: Tests GitHub API integration for release fetching (when DDEV_RUN_GET_TESTS is set)
- **CircularDependencyDetection**: Unit tests for circular dependency detection logic

### Test Infrastructure

- All tests use realistic addon structures with proper install.yaml files
- Tests verify both successful installations and expected failure cases  
- Comprehensive cleanup ensures tests don't interfere with each other
- Tests validate file creation, manifest generation, and proper addon removal

### Existing Test Compatibility

- All existing addon tests continue to pass without modification
- Test suite runs in both cmd and pkg contexts
- Integration with existing test infrastructure (TestSites, DdevBin, etc.)

### New Features Added

- **Automatic dependency installation**: Default behavior change - dependencies now install automatically
- **Runtime dependency discovery**: New capability for dynamic dependency detection
- **--skip-deps flag**: New CLI option for backward compatibility

### Configuration Changes

- **Runtime dependency files**: New `.runtime-deps-{addon-name}` files created in `.ddev/` during installation
- **Enhanced manifests**: Dependency information tracked in addon manifests for proper removal

### User Impact

- **Improved UX**: Users no longer need to manually track and install addon dependencies
- **Backward Compatible**: Existing workflows continue to work unchanged
- **Opt-out Available**: `--skip-deps` flag provides escape hatch for edge cases

### Documentation Updates

- Updated `docs/content/users/extend/creating-add-ons.md` Dependencies section
- Enhanced `docs/content/users/usage/commands.md` for `ddev add-on get` command
- Added information about automatic installation and the `--skip-deps` flag

### Migration Notes

-  Add-ons with dependencies must declare them with a resolvable format: `ddev/ddev-redis` instead of `ddev-redis` or `redis`.
- **Enhanced functionality**: Addons with dependencies in install.yaml will now install dependencies automatically
- **New addon capabilities**: Addon developers can implement runtime dependency discovery

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Comment by rfay** [comment]
Edited review from GitHub Copilot:

PR review: feat(add-ons): add-on dependencies should be automatically downloaded fixes #5337

Key findings and recommendations
1) Incomplete install flow for GitHub dependencies (critical)
- InstallAddonFromTarball parses install.yaml and installs dependencies, but stops short of the full install (pre/post actions, project files, manifest updates), yet prints success.
- Dependencies installed via the “owner/repo” path may appear “installed” without actually applying files/actions.
- Recommendation: Complete InstallAddonFromTarball to run the same full path as the directory-based installer (or centralize a single “install from extractedDir” routine used by both). Add a test that installs a dependency via GitHub to catch this.

2) Manifest key vs dependency identifier mismatches
- InstallDependencies checks m[dep], but dep can be owner/repo, path, or URL. If GatherAllManifests doesn’t key by all canonical forms (name, owner/repo, absolute path), already-installed addons may be reinstalled.
- Recommendation: Normalize identifiers and/or expand manifest keys. Add tests mixing already-installed GitHub and local-path dependencies to prove idempotency.

3) Duplication/drift between cmd and ddevapp logic
- Dependency resolution (relative path handling, runtime deps) exists in both cmd/addon-get and pkg/ddevapp.
- Recommendation: Keep cmd thin; centralize install/dependency logic in ddevapp and call from cmd.

4) Circular dependency detection normalization
- The stack detects cycles and prints a helpful chain. However, identifiers differ by format (absolute path vs relative vs owner/repo), which could obscure cycles if mixed.
- Recommendation: Normalize the identifier placed on the stack (e.g., absolute path for local, owner/repo for GitHub). Add a test that mixes owner/repo with relative path in a cycle.

5) ProcessAddonAction and extractedDir access
- Directory-based install runs actions but doesn’t pass extractedDir explicitly. Tests include an “extracted-dir-addon” pattern; ensure actions can reliably access co-located files.
- Recommendation: Provide extractedDir context to action processing or codify working directory expectations. Add a test that verifies copying/using files from the extracted add-on dir.

6) Imports/build hygiene
- New functions use context and go-github types. Double-check imports compile cleanly in pkg/ddevapp with the newly added code.

7) Output verbosity
- util.Success logs are helpful but can be chatty. Ensure noisy diagnostics are gated behind --verbose.

8) Tests to add
- A dependency graph that includes a GitHub dependency (not only local dir) verifying full install behavior (files/actions), which should fail prior to completing InstallAddonFromTarball.
- Mixed dependencies (GitHub + local), proving no redundant installs.
- A mixed-format circular chain (owner/repo + relative path).
- Runtime deps file that includes a GitHub repo dep.

9) Documentation
- Docs are good; consider explicitly noting:
  - Relative path resolution is relative to the extracted add-on directory.
  - Runtime deps file lifecycle (.runtime-deps-{addon} is removed after install).

10) Repo hygiene
- CLAUDE.md update references “Task Master AI Instructions” and ./.taskmaster/CLAUDE.md; and .gitignore additions for /.mcp.json and /.env.example. These look like local tooling artifacts; consider removing unless they’re meant to be part of the repo.

Behavior/CLI correctness
- --no-dependencies: The check for relative deps reading install.yaml to get a “name” is good. Ensure manifest key matching (point #2) to avoid false negatives with GitHub/URL deps.
- Runtime deps: Parsing, install, and cleanup look sound; verbose mode is useful for debugging.

**Comment by rfay** [comment]
I ran [script](https://gist.github.com/rfay/c1c6f31231c5411d44a899feaa2fec56) against all public add-ons.

## Official/important
- [x] ddev/ddev-redis-commander fails, this is probably a serious issue
- [x] ddev-platformsh fails to install, but I think because it can't with no config
- [x] stasadev/ddev-redis-insight fails to install, probably because of a bug here

## Dependency specification problems
- [x] 2ndkauboy/ddev-elasticvue fails due to inadequate dependency https://github.com/2ndkauboy/ddev-elasticvue/pull/2
- [x] ckng/ddev-directus-postgres (same) https://github.com/ckng/ddev-directus-postgres/pull/1
- [x] d34dman/ddev-notification-server (same) https://github.com/d34dman/ddev-notification-server/pull/1
- [x] froboy/ddev-tsh (same) https://github.com/froboy/ddev-tsh/pull/2
- [x] JanoPL/ddev-kibana https://github.com/JanoPL/ddev-kibana/pull/36
- [x] Metadrop/ddev-selenium-video https://github.com/Metadrop/ddev-selenium-video/pull/3

**Comment by rfay** [comment]
Another round of Copilot review has some decent suggestions

Key items:

- [x] Is there really a mention of `ddev add-on install` ? Fix
- [x] Supporting $GITHUB_TOKEN/$GH_TOKEN is really a decent thing although actually outside the scope of this.
- [x] "Documentation consistency" section is worthy.
- [x] "CLI UX" suggestions seem valid, giving better info to people.
- [x] Dependency syntax needs to be stated in docs

## High-priority feedback

- CLI naming consistency
  - The PR body alternates between “ddev add-on get” and “ddev add-on install <add-on> --no-dependencies”. Today’s command is “get”. Ensure the flag is added to the actual command and docs are consistent. If “install” is an alias, pick one canonical verb to document.
  - Consider a short form for the flag: support both `--no-dependencies` and `--no-deps`, and possibly `-D`.

- Dependency tracking and removal semantics
  - Manifests should record “who required what” so removal behaves correctly:
    - If A depends on B and C also depends on B, removing A must not remove B while C remains installed.
    - If B was user-installed before A, removing A must not remove B.
  - Please document how you record:
    - Direct vs transitive dependencies
    - “User-installed” vs “auto-installed” origin
    - Dependents (“required-by” list)

- Versioning and conflict handling
  - Define how versions/refs are specified and resolved for GitHub add-ons (e.g., `org/repo@v1.2.3`, tag, SHA).
  - If two add-ons require the same dependency at different versions, what happens? Fail with a clear error? Choose a policy and document it.
  - If default is “latest release” without pinning, state this clearly and encourage pinning.

- Runtime dependency discovery safety
  - Ensure `.runtime-deps-{addon-name}` files are:
    - Created only under `.ddev/` of the active project
    - Only read for the add-on undergoing install
    - Cleaned up deterministically after success or failure
    - Sanitized: trim whitespace, ignore comments and blank lines, deduplicate entries
  - Add tests for malformed input, CRLF handling, duplicate entries, and path traversal attempts.

- Relative path and symlink security
  - Decide and document whether absolute paths are allowed. If allowed, guard against escapes outside the project root unless explicitly intended.
  - Resolve symlinks and fail if the target escapes project boundaries (unless explicitly supported).
  - Add tests for traversal (`../`) and symlink edge cases.

- Error messages and diagnostics
  - For circular dependencies, print the cycle chain (e.g., `A -> B -> C -> A`).
  - Before installation, print a plan (resolved dependency graph and the exact set to be installed). Consider `--dry-run` to show the plan only.
  - When resolution or fetch fails, provide actionable messages: what was attempted, next steps, and a link to docs.

- GitHub API integration resilience
  - Add retry with exponential backoff on transient failures.
  - Respect `GITHUB_TOKEN`/`GH_TOKEN` for authentication and document rate-limit behavior. Consider GitHub Enterprise host support.
  - Consider ETag/If-None-Match caching to reduce rate limiting.
  - If possible, provide a fallback (e.g., Git clone) when API calls fail but the repo is fetchable; otherwise document API requirement.

- Documentation consistency and completeness
  - Creating add-ons:
    - Specify dependency declaration syntax with examples:
      - GitHub (`org/repo`), with and without explicit versions
      - Local/relative paths
      - Dynamic dependencies: exact filename pattern `.runtime-deps-{addon}`, supported comment syntax, lifecycle (created and removed), and security considerations
  - Commands:
    - Ensure command/flag names match the actual CLI
    - Clearly highlight the default behavior change (auto-install on by default) and how to opt out
  - Migration notes:
    - Detection/error when dependencies are non-resolvable slugs (e.g., `redis` instead of `org/repo`)
    - Guidance to migrate to resolvable forms

## Test scenarios to add

- Shared dependency removal
  - Remove one parent of a shared dependency; ensure the shared leaf remains until all parents are removed.
  - Remove the leaf explicitly; ensure parents remain and the manifest updates correctly.

- Idempotency
  - Repeated “get” of the same add-on with dependencies already present should be a no-op with stable output.

- Windows/CRLF handling
  - `.runtime-deps` parsing on Windows with CRLF line endings.

- Network/rate limiting
  - Mock transient errors and rate limits for `GetGitHubRelease` and verify retries/fail messages.

- Malformed `install.yaml`
  - Unresolvable dependencies and invalid formats produce clear, actionable errors.

- Large graphs
  - Depth and breadth to validate deterministic topological ordering and performance/UX.

## Design and implementation suggestions

- Topological ordering
  - Ensure deterministic ordering and consistent output across runs.
  - Prefer sequential installs now; if you later add parallelism, document ordering guarantees.

- State/manifest design
  - Prefer a single source of truth that includes:
    - Installed add-on name, origin (user vs auto), version/ref, installed timestamp
    - Dependents (“required-by”) and direct dependencies (with exact ref used)
  - Unit-test merges when an already-installed add-on later becomes a dependency of another.

- CLI UX
  - Plan output example:
    - “Will install: A, B@v1.2.3, C (path: ../local-addon)”
    - “Already installed: D”
    - “Skipped (per --no-dependencies): E, F”
  - Consider `ddev add-on info <name>` to display dependencies/dependents from the manifest for troubleshooting.

- Security and supply chain
  - Consider checksums or signatures for downloaded assets where applicable.
  - Document that pre/post-install hooks and runtime dependency execution can run untrusted code; provide best-practice guidance.

- Logging and verbosity
  - `-v` should print high-level API interactions, resolution steps, and write/remove operations.
  - Keep default output succinct while clearly stating what was installed and why.

## Minor nits and polish

- Naming of runtime deps file
  - `.runtime-deps-{addon-name}` is fine; consider `.runtime-deps.{name}` for readability. Pick one and document.

- Help text and examples
  - Ensure help output and docs reflect the new default behavior and flags.
  - Provide examples with tags and local paths, plus a dynamic dependency example.

## Release notes

- Highlight default behavior change prominently: dependencies are now installed automatically.
- Provide a short snippet to restore prior behavior with `--no-dependencies` (and any env var override if added).
- Link directly to:
  - “Creating add-ons” dependency section
  - Runtime dependencies section
  - Command reference for `ddev add-on get`

## Questions to clarify

- What is the exact dependency syntax in `install.yaml`? Is `org/repo@vX.Y.Z` (or tag/SHA) supported?
- How are conflicts handled when two add-ons require different versions of the same dependency?
- Are absolute paths supported? If yes, what guardrails prevent escapes outside the project?
- Will there be an environment variable to globally opt out (e.g., CI or air‑gapped environments)?
- Does `GetGitHubRelease` support GitHub Enterprise hosts and custom tokens?

**Comment by rfay** [comment]
I'm sure there's more work to be done, but think it's feature complete. Thanks for taking a look.

File: pkg/ddevapp/addons.go [hunk 1]
```diff
@@ -802,6 +831,591 @@ func RemoveAddon(app *DdevApp, addonName string, verbose bool, skipRemovalAction
...
+		}
+		if !releaseFound {
+			return "", "", fmt.Errorf("no release found for %v with tag %v", repo, requestedVersion)
+		}
+	}
+
+	tarballURL = releases[releaseItem].GetTarballURL()
+	downloadedRelease = releases[releaseItem].GetTagName()
+	return tarballURL, downloadedRelease, nil
+}
```
Comment (stasadev) [inline pkg/ddevapp/addons.go]: Let's move this function to `pkg/github/github.go`, where it should belong, and return `releases[releaseItem]` instead to make it more flexible.

Comment (rfay) [reply pkg/ddevapp/addons.go]: Moved, thanks!

**Comment by rfay** [comment]
Changed to `--skip-deps`, good suggestion and significant improvement I think.

Current version is working with ddev-upsun.

File: cmd/ddev/cmd/addon-get_test.go [hunk 2]
```diff
@@ -388,3 +388,172 @@ func TestCmdAddonGetWithDotEnv(t *testing.T) {
...
+	// Verify runtime dependency installation
+	require.Contains(t, out, "Installing runtime dependencies")
+	require.Contains(t, out, "Successfully installed mock-redis from directory")
+	require.Contains(t, out, "Successfully installed mock-redis-commander from directory")
+	require.Contains(t, out, "Runtime dependencies addon configured successfully")
+
+	// Verify files were created by all addons
+	require.FileExists(t, app.GetConfigPath("docker-compose.redis.yaml"))
+	require.FileExists(t, app.GetConfigPath("docker-compose.redis-commander.yaml"))
+	require.FileExists(t, app.GetConfigPath("config.yaml"))
```
Comment (stasadev) [inline cmd/ddev/cmd/addon-get_test.go]: 1. Having `config.yaml` with `#ddev-generated` in test doesn't test anything, simply because it can't be installed into existing DDEV project:

```
NOT overwriting /path/to/project/.ddev/config.yaml. The #ddev-generated signature was not found in the file, so it will not be overwritten. You can remove the file and use ddev add-on get again if you want it to be replaced: signature was not found in file /path/to/project/.ddev/config.yaml
```

This should be replaced with `config.something.yaml` or `foobar-config.yaml`

(I left six comments in total for this, six places to update)

File: cmd/ddev/cmd/addon-get_test.go [hunk 3]
```diff
@@ -388,3 +388,172 @@ func TestCmdAddonGetWithDotEnv(t *testing.T) {
...
+	// Verify runtime dependency installation
+	require.Contains(t, out, "Installing runtime dependencies")
+	require.Contains(t, out, "Successfully installed mock-redis from directory")
+	require.Contains(t, out, "Successfully installed mock-redis-commander from directory")
+	require.Contains(t, out, "Post-install runtime dependencies file created")
+
+	// Verify files were created by all addons
+	require.FileExists(t, app.GetConfigPath("docker-compose.redis.yaml"))
+	require.FileExists(t, app.GetConfigPath("docker-compose.redis-commander.yaml"))
+	require.FileExists(t, app.GetConfigPath("config.yaml"))
```
Comment (stasadev) [inline cmd/ddev/cmd/addon-get_test.go]: 2

File: cmd/ddev/cmd/testdata/TestAddonWithRuntimeDeps/post_install_runtime_deps_addon/config.yaml
Comment (stasadev) [inline cmd/ddev/cmd/testdata/TestAddonWithRuntimeDeps/post_install_runtime_deps_addon/config.yaml:1]: 3

File: cmd/ddev/cmd/testdata/TestAddonWithRuntimeDeps/post_install_runtime_deps_addon/install.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+name: post-install-runtime-deps-addon
+description: Test addon that creates runtime dependencies during post-install actions
+
+pre_install_actions:
+  - |
+    #ddev-description:Pre-install setup
+    echo "Pre-install actions for post-install runtime deps addon"
+
+project_files:
+  - config.yaml
```
Comment (stasadev) [inline cmd/ddev/cmd/testdata/TestAddonWithRuntimeDeps/post_install_runtime_deps_addon/install.yaml]: 4

File: cmd/ddev/cmd/testdata/TestAddonWithRuntimeDeps/runtime_deps_addon/config.yaml
Comment (stasadev) [inline cmd/ddev/cmd/testdata/TestAddonWithRuntimeDeps/runtime_deps_addon/config.yaml:1]: 5

File: cmd/ddev/cmd/testdata/TestAddonWithRuntimeDeps/runtime_deps_addon/install.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  - |
+    #ddev-description:Create runtime dependencies file
+    echo "# Runtime dependencies discovered during pre-install" > .ddev/.runtime-deps-runtime-deps-addon
+    echo "../../TestRealWorldDependencies/mock_redis" >> .ddev/.runtime-deps-runtime-deps-addon
+    echo "" >> .ddev/.runtime-deps-runtime-deps-addon
+    echo "# This is a comment" >> .ddev/.runtime-deps-runtime-deps-addon
+    echo "../../TestRealWorldDependencies/mock_redis_commander" >> .ddev/.runtime-deps-runtime-deps-addon
+
+project_files:
+  - config.yaml
```
Comment (stasadev) [inline cmd/ddev/cmd/testdata/TestAddonWithRuntimeDeps/runtime_deps_addon/install.yaml]: 6

File: cmd/ddev/cmd/testdata/TestRealWorldDependencies/mock_redis/docker-compose.redis.yaml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#ddev-generated
+# Mock Redis Docker Compose configuration
+services:
+  redis:
+    image: redis:alpine
+    container_name: ddev-${DDEV_SITENAME}-redis
+    ports:
```
Comment (stasadev) [inline cmd/ddev/cmd/testdata/TestRealWorldDependencies/mock_redis/docker-compose.redis.yaml]: It's mocking, so exposing ports not needed:

```suggestion
    expose:
```

File: cmd/ddev/cmd/testdata/TestRealWorldDependencies/mock_redis_commander/docker-compose.redis-commander.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#ddev-generated
+# Mock Redis Commander Docker Compose configuration
+services:
+  redis-commander:
+    image: rediscommander/redis-commander
+    container_name: ddev-${DDEV_SITENAME}-redis-commander
+    ports:
```
Comment (stasadev) [inline cmd/ddev/cmd/testdata/TestRealWorldDependencies/mock_redis_commander/docker-compose.redis-commander.yaml]:
```suggestion
    expose:
```

File: cmd/ddev/cmd/testdata/TestRealWorldDependencies/mock_redis_commander/docker-compose.redis-commander.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+      - "8081"
+    environment:
+      - REDIS_HOSTS=redis:redis:6379
```
Comment (stasadev) [inline cmd/ddev/cmd/testdata/TestRealWorldDependencies/mock_redis_commander/docker-compose.redis-commander.yaml]: Just to make it look real:
```suggestion
      - REDIS_HOST=redis
```

File: pkg/versionconstants/versionconstants.go [hunk 9]
```diff
@@ -1,7 +1,15 @@
...
+import (
+	"os"
+	"runtime/debug"
+	"strings"
+)
+
+// DdevVersion is the current version of DDEV. Normally set via -ldflags from the Makefile
+// using `git describe --tags --always --dirty`. If not provided, we derive a best-effort
+// value. Prefer VERSION env var, otherwise use Go build-info hash.
+var DdevVersion = "" // Note that this is overridden by make
```
Comment (stasadev) [inline pkg/versionconstants/versionconstants.go:13]: Some tests still have `v0.0.0-overridden-by-make` in conditions, which is likely to affect them under certain conditions (i.e. when running them inside GoLand):

- TestDebugMatchConstraintCmd
- TestCmdVersion
- TestCmdAddonDdevVersionConstraint (inside install.yaml)

`v0.0.0-overridden-by-make` should be replaced with something else or removed?

Comment (rfay) [reply pkg/versionconstants/versionconstants.go:13]: I apologize for even fiddling with this here, but obviously I've been long annoyed by it. I fiddled with those tests and made a fallback for the version creation. It's sad to add code for this situation in versionconstants.go, but OTOH it should very rarely get executed, and only in developer context.
